### ISSUE TITLE
Basket of issues on version of 20220122

### ISSUE THREAD

**Issue by bryn51** [bryn51/smartcitizen-kit-21-WIP#6]
I observe the following issues from the UI screen on the current version published today.

(a) SCD4x Co2 reporting ridiculous values for CO2 concentration.  the calls to sparkfun_SCD4x.setAmbientPressure() and sparkfun_SCD4x.performForcedRecalibration appear to have no effect.  automatic self calibration is in operation; but this can take some time to work;  It only works when the device is continuously powered for a long period of time; Which is not possible under current "Taichung development Lab"  conditions; as frequent power cycles occur due to correction of other issues: see below.

(b) Hang after power cycle, and after config details are entered.  The device just hangs; with no diagnostic.  If the reset button or reset command are entered then the hang does not occur.  When the device is reset after this hang; the config details appear to have been stored;

(c) hang after some time (tens of minutes ...) has elapsed following a reset(); with config details previously entered and stored.

(d) Second OLED does not display anything.  With the code included in this release; the 2nd OLED should display readings in the same cycle as first OLED; but it is not working.  2nd OLED does not even display the SC logo at the start.  I suspect this might be I2C addressing concerns; investigation is proceeding.

(f) during Phase 2 (getReading) cycle this appears:  (All the other entries appear only once).  Notice the time difference between first appearance and the other two.

12:19:15.050 -> Light: 9 Lux
12:19:15.219 -> Light: 9 Lux
12:19:15.219 -> Light: 9 Lux

**Comment by bryn51** [comment]
This has been parially addressed by the following:

(a) SCD41 anomalous high readings for CO2.  I thought perhaps the device is faulty, so I purchased a replacement.  Results are much the same, thus pointing to operating algorithm as the root cause.  I have tried entering commands to 'Force' recalibration to occur.  the documentation indicates that the readings should move towards the forced figure, but its not happening.  Needs further work.

(b) Hang after config changed adding network details. I altered the code to prompt the user to perform a reset (button) after config has changed.  Seems to do the trick.

(c) With GPS disabled; INA219 disconnected there are no more hangs.  The system runs for hours on end.

(d) I discarded the 2nd OLED: it appears to be faulty (I tested it using a standalone arduino board).

(f) This no longer occurs; I did nothing special to fix it.  I cannot trace the cause unless it starts happening again.
